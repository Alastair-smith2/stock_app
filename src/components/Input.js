import React, { Component } from "react";
import { View, Text, TextInput, StyleSheet } from "react-native";

export default class Input extends Component {
  onTextChange = props => {
    this.props.onTextInput(this.props.type, props);
  };

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.label}>{this.props.label}</Text>
        <TextInput
          value={this.props.value}
          placeholder={`Please enter you ${this.props.type}`}
          onChangeText={text => this.onTextChange(text)}
          type={this.props.type}
          underlineColorAndroid={"transparent"}
          autoCapitalize={"none"}
          style={styles.textInput}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignContent: "center"
  },
  label: {
    fontSize: 16,
    fontFamily: "Futura"
  },
  textInput: {
    flex: 1.5,
    height: 40,
    borderColor: "#BBB",
    borderWidth: 1,
    padding: 10
  }
});
