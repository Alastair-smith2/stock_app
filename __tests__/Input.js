import React from "react";
import { shallow } from "enzyme";

import Input from "../src/components/Input.js";

describe("The Input component", () => {
  let wrapper;
  let onTextInput = jest.fn();
  beforeEach(() => {
    wrapper = shallow(
      <Input
        value=""
        type="username"
        onTextInput={onTextInput}
        placeholder="Please enter your username"
      />
    );
    onTextInput.mockReset();
  });
  describe("It should render", () => {
    it("Should render", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("Features", () => {
    it("Should have 1 text input box", () => {
      expect(wrapper.find("TextInput")).toHaveLength(1);
    });

    it("Should have a prop called onTextInput that is called via onTextChange", () => {
      // the type is passed via a prop, therefore only the value of the text needs to be passed through
      wrapper.instance().onTextChange("Alastair");
      expect(onTextInput.mock.calls.length).toEqual(1);
      expect(onTextInput.mock.calls[0][0]).toEqual("username");
      expect(onTextInput.mock.calls[0][1]).toEqual("Alastair");
    });

    it("Should have a function that handles the input change of TextInput", () => {
      wrapper.find("TextInput").simulate("changeText", "test");
      expect(onTextInput.mock.calls.length).toEqual(1);
      expect(onTextInput.mock.calls[0][1]).toEqual("test");
    });

    it("Should have a secureTextProperty that exists when the type passed through is password", () => {
      wrapper.setProps({
        type: "password",
        placeholder: "Please enter your password"
      });
      expect(wrapper.find("TextInput").prop("secureTextEntry")).toEqual(true);
      expect(wrapper.find("TextInput").prop("placeholder")).toEqual(
        "Please enter your password"
      );
    });
  });
});
