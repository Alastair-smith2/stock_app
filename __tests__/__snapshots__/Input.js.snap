// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Input component It should render Should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    onTextInput={[MockFunction]}
    type="username"
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontFamily": "Futura",
              "fontSize": 16,
            }
          }
        />,
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="Please enter you username"
          style={
            Object {
              "borderColor": "#BBB",
              "borderWidth": 1,
              "flex": 1.5,
              "height": 40,
              "padding": 10,
            }
          }
          type="username"
          underlineColorAndroid="transparent"
          value=""
        />,
      ],
      "style": Object {
        "alignContent": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": undefined,
          "ellipsizeMode": "tail",
          "style": Object {
            "fontFamily": "Futura",
            "fontSize": 16,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowFontScaling": true,
          "autoCapitalize": "none",
          "onChangeText": [Function],
          "placeholder": "Please enter you username",
          "style": Object {
            "borderColor": "#BBB",
            "borderWidth": 1,
            "flex": 1.5,
            "height": 40,
            "padding": 10,
          },
          "type": "username",
          "underlineColorAndroid": "transparent",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontFamily": "Futura",
                "fontSize": 16,
              }
            }
          />,
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            onChangeText={[Function]}
            placeholder="Please enter you username"
            style={
              Object {
                "borderColor": "#BBB",
                "borderWidth": 1,
                "flex": 1.5,
                "height": 40,
                "padding": 10,
              }
            }
            type="username"
            underlineColorAndroid="transparent"
            value=""
          />,
        ],
        "style": Object {
          "alignContent": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": undefined,
            "ellipsizeMode": "tail",
            "style": Object {
              "fontFamily": "Futura",
              "fontSize": 16,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": true,
            "autoCapitalize": "none",
            "onChangeText": [Function],
            "placeholder": "Please enter you username",
            "style": Object {
              "borderColor": "#BBB",
              "borderWidth": 1,
              "flex": 1.5,
              "height": 40,
              "padding": 10,
            },
            "type": "username",
            "underlineColorAndroid": "transparent",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
